### PR TITLE
use nodes; clean up code (.net-style)

### PR DESCRIPTION
**Pull request by gregsdennis**
Not sure if you have a build on this project or not.  The running instructions are in unix/mac, so I didn't run it locally, but I know it compiles.

### CODE REVIEW

**Comment by gregsdennis** [comment]
looks like I need to add a .gitignore for this folder.

**Comment by Julian** [comment]
Nice, thanks!

> Not sure if you have a build on this project or not.

(There's a build, though it's got a flaky test in CI which I haven't spent time fixing yet as it'll get better when I change how output is collected and it passes locally).

>  The running instructions are in unix/mac

Yeah, Windows docs also obviously are missing... What I get locally from running this with some (bowtie-less) test input is:

```
⊙  podman run --rm -i bowtie/dotnet-json-everything <<EOF                                                                                                                         julian@Airm ●
{"cmd": "start", "version": 1}
{"cmd": "run", "seq": 0, "case": {"description": "stuff", "schema": {"const": {"foo": "bar"}}, "tests": [{"description": "a test", "instance": {"foo": "bar"}, "valid": true}, {"description": "another test", "instance": {}, "valid": false}]}}
{"cmd": "stop"}
EOF
{
  "ready": true,
  "version": 1,
  "implementation": {
    "language": "dotnet",
    "name": "json-everything",
    "homepage": "https://json-everything.net/json-schema/",
    "issues": "https://github.com/gregsdennis/json-everything/issues"
  }
}
Unhandled exception. System.NullReferenceException: Object reference not set to an instance of an object.
   at Program.<Main>$(String[] args) in /source/Program.cs:line 51
```

Any chance you see what that comes from?

**Comment by gregsdennis** [comment]
I'll have a look

**Comment by Julian** [comment]
Found it, think it's all good now thanks for the PR!

File: implementations/dotnet-json-everything/Program.cs [hunk 1]
```diff
@@ -58,7 +58,7 @@
 
 			var runResult = new JsonObject
 			{
-				["seq"] = root["seq"],
+				["seq"] = root["seq"].GetValue<int>(),
```
Comment (gregsdennis) [inline implementations/dotnet-json-everything/Program.cs:61]: This just gets cast back to a `JsonNode` anyway.  There's no benefit to extracting the `int` except for the side effect of checking that it is one.

File: implementations/dotnet-json-everything/Program.cs [hunk 2]
```diff
@@ -41,7 +41,7 @@
 			}
 
 			var testCase = root["case"];
-			var schemaText = root["schema"];
+			var schemaText = testCase["schema"];
```
Comment (gregsdennis) [inline implementations/dotnet-json-everything/Program.cs:44]: Ha. oops.

File: implementations/dotnet-json-everything/Program.cs (hunk 1 above)
Comment (Julian) [reply implementations/dotnet-json-everything/Program.cs:61]: Without it I got another error. One that said:

```
Unhandled exception. System.InvalidOperationException: The node already has a parent.
   at System.Text.Json.ThrowHelper.ThrowInvalidOperationException_NodeAlreadyHasParent()
   at System.Text.Json.Nodes.JsonNode.AssignParent(JsonNode parent)
   at System.Text.Json.Nodes.JsonObject.<>c__DisplayClass9_0.<SetItem>b__0()
   at System.Text.Json.JsonPropertyDictionary`1.SetValue(String propertyName, T value, Action assignParent)
   at System.Text.Json.Nodes.JsonObject.SetItem(String propertyName, JsonNode value)
   at Program.<Main>$(String[] args) in /source/Program.cs:line 59
```

Is there another way that should have been handled? (Google suggested a yet sillier one to serialize back to a string and deserialize again)

Comment (gregsdennis) [reply implementations/dotnet-json-everything/Program.cs:61]: Hmm yeah, I understand that, but I wouldn't have expected it.

Nodes hold a reference to the parent, which means they can only exist in a single tree.  Not sure how else to handle that.

**Pull request merged**